e2e/_suites/kubernetes-autodiscover/kubernetes_test.go

```diff
@@ -135,11 +135,17 @@ func (c *kubernetesCluster) initialize(ctx context.Context) error {
 	name := "kind-" + uuid.New().String()
 	c.kubeconfig = filepath.Join(c.tmpDir, "kubeconfig")
 
-	output, err := shell.Execute(ctx, ".", "kind", "create", "cluster",
+	args := []string{
+		"create", "cluster",
 		"--name", name,
 		"--config", "testdata/kind.yml",
 		"--kubeconfig", c.kubeconfig,
-	)
+	}
+	if version, ok := os.LookupEnv("KUBERNETES_VERSION"); ok && version != "" {
```

Comment:
I guess this is needed to install the k8s version no matter whether the worker has it or not, right? Does it makes sense to install it instead of failing?

Reply:
Each version of kind uses a version of kubernetes by default, it is ok not to set this option. With this option we are adding support for selecting a different version. I can rephrase of remove the log message below if it is confusing.

Reply:
I'm OK with enriching the log explaining the purpose :)